devenv/lib/volta.py

```diff
@@ -23,49 +23,70 @@ class UnexpectedPlatformError(Exception):
     pass
 
 
-def install_volta(unpack_into: str) -> None:
+def determine_platform(unpack_into: str) -> str | None:
     system = platform.system()
     if system == "Linux":
         if platform.machine() == "x86_64":
-            name = f"volta-{_version}-linux.tar.gz"
+            return f"volta-{_version}-linux.tar.gz"
         else:
             proc.run(("brew", "install", "volta"), exit=True)
             proc.run(
                 ("sh", "-c", f"ln -sfn {homebrew_bin}/volta* {unpack_into}/"),
                 exit=True,
             )
-            return
+            return None
     elif system == "Darwin":
         suffix = "-aarch64" if platform.machine() == "arm64" else ""
-        name = f"volta-{_version}-macos{suffix}.tar.gz"
+        return f"volta-{_version}-macos{suffix}.tar.gz"
     else:
         raise UnexpectedPlatformError(f"Unexpected OS: {platform.platform()}")
 
-    url = (
+
+def build_url(name: str) -> str:
+    return (
         "https://github.com/volta-cli/volta/releases/download/"
         f"v{_version}/{name}"
     )
 
+
+def download_and_unpack_archive(name: str, unpack_into: str) -> None:
+    url = build_url(name)
+
     archive_file = archive.download(url, _sha256[name])
     archive.unpack(archive_file, unpack_into)
 
 
-def install() -> None:
-    unpack_into = f"{root}/bin"
+def install_volta(unpack_into: str) -> None:
+    name = determine_platform(unpack_into)
+    if name is None:
+        return
+    download_and_unpack_archive(name, unpack_into)
+
 
-    if (
-        which("volta", path=unpack_into) == f"{unpack_into}/volta"
+def are_volta_and_node_installed(unpack_into: str) -> bool:
```

Comment:
i think you've overdone it with breaking code up into too many small functions... this piece of logic is straightforward enough it doesn't warrant the overhead of another function call + a whole test file

Reply:
the other functions are fine, i'm just proposing you inline this one back